NewDawn/NewDawn/MainPageViewController.swift

```diff
@@ -8,41 +8,108 @@
-    
-    func loadProfileIndex() -> Int {
-        if let index = self.localReadKeyValue(key: MAIN_PAGE_PROFILE_INDEX) as? Int {
-            return index
-        }
-        self.localStoreKeyValue(key: MAIN_PAGE_PROFILE_INDEX, value: 0)
-        return 0
-    }
-
-    func updateProfileIndex() -> Void {
-        self.localStoreKeyValue(key: MAIN_PAGE_PROFILE_INDEX, value: loadProfileIndex() + 1)
-    }
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        if loadProfileIndex() >= USER_DUMMY_DATA.count {
-            // TODO: When the user has seen all profiles, we go back to the first profile.
-            // In the future, we should go to an ending page
-            self.localStoreKeyValue(key: MAIN_PAGE_PROFILE_INDEX, value: 0)
+        self.getUserData(){ () -> () in
+            if ProfileIndexUtil.loadProfileIndex() >= USER_DUMMY_DATA.count {
+                // TODO: When the user has seen all profiles, we go back to the first profile.
+                // In the future, we should go to an ending page
+                self.localStoreKeyValue(key: MAIN_PAGE_PROFILE_INDEX, value: 0)
+            }
+            print(USER_DUMMY_DATA)
+            viewModel = MainPageViewModel(userProfile: UserProfile(data: USER_DUMMY_DATA[ProfileIndexUtil.loadProfileIndex()]))
+            tableView.dataSource = viewModel
+            tableView.delegate = viewModel
+            tableView.rowHeight = UITableView.automaticDimension
+            tableView.estimatedRowHeight = UITableView.automaticDimension
+            navigationItem.hidesBackButton = true
         }
-        viewModel = MainPageViewModel(userProfile: UserProfile(data: USER_DUMMY_DATA[loadProfileIndex()]))
-        tableView.dataSource = viewModel
-        tableView.delegate = viewModel
-        tableView.rowHeight = UITableView.automaticDimension
-        tableView.estimatedRowHeight = UITableView.automaticDimension
-        navigationItem.hidesBackButton = true
     }
-
     @IBAction func skipButtonTapped(_ sender: Any) {
-        updateProfileIndex()
+        // The profile is skipped
+        if ProfileIndexUtil.reachLastProfile() {
+            self.performSegue(withIdentifier: "mainPageEnd", sender: nil)
+        } else {
+            ProfileIndexUtil.updateProfileIndex()
+            self.performSegue(withIdentifier: "mainPageSelf", sender: nil)
+        }
+    }
+    
+    func getUserData(handleComplete:(()->())) {
+        let request = createGetProfileRequest()
+        processSessionTasks(request: request!, callback: readReponse)
+        handleComplete()
+    }
+    
+    func readReponse(response: NSDictionary) -> Void{
+        let profile_responses = response["objects"] as? [[String: Any]]
+        for profile in profile_responses!{
+            let dummy_user = getProfileInfo(profile: profile)
+            USER_DUMMY_DATA.append(dummy_user as NSDictionary)
+        }
+    }
+    
+    func encodeParams(raw_params: Dictionary<String, String>) -> String{
```

Comment:
In HttpUtil.swift, create a class called HttpUtil and put this function there. You can then call HttpUtil.encodeParams() anywhere

Reply:
However, I see HttpUtil inherits from UIImageView though. Can I still do that?